evap/results/views.py

```diff
@@ -224,16 +225,35 @@ def remove_textanswers_that_the_user_must_not_see(evaluation_result, user, repre
                     answer for answer in question_result.answers
                     if can_textanswer_be_seen_by(user, represented_users, answer, view)
                 ]
+            if isinstance(question_result, RatingResult) and question_result.additional_text_result:
+                question_result.additional_text_result.answers = [
+                    answer for answer in question_result.additional_text_result.answers
+                    if can_textanswer_be_seen_by(user, represented_users, answer, view)
+                ]
         # remove empty TextResults
-        questionnaire_result.question_results = [
-            result for result in questionnaire_result.question_results
-            if not isinstance(result, TextResult) or len(result.answers) > 0
-        ]
+        cleaned_results = []
+        for result in questionnaire_result.question_results:
+            if isinstance(result, TextResult):
+                if result.answers:
+                    cleaned_results.append(result)
+            elif isinstance(result, HeadingResult):
+                cleaned_results.append(result)
+            else:
+                if result.additional_text_result and not result.additional_text_result.answers:
+                    result.additional_text_result = None
+                cleaned_results.append(result)
+        questionnaire_result.question_results = cleaned_results
 
 
 def filter_text_answers(evaluation_result):
     for questionnaire_result in evaluation_result.questionnaire_results:
-        questionnaire_result.question_results = [result for result in questionnaire_result.question_results if isinstance(result, TextResult)]
+        question_results = []
+        for result in questionnaire_result.question_results:
+            if isinstance(result, TextResult):
+                question_results.append(result)
+            elif result.additional_text_result:
```

Comment:
why don't you need to check for `isinstance(RatingResult)` before accessing `additional_text_result`?